File: TransCoder_tokenized_test_set_functions/java/MAXIMUM_SEGMENT_VALUE_PUTTING_K_BREAKPOINTS_NUMBER.java
static int findMaxSegment ( String s, int k ) {
  int seg_len = s . length ( ) - k;
  int res = 0;
  for ( int i = 0;
  i < seg_len;
  i ++ ) res = res * 10 + ( s . charAt ( i ) - '0' );
  int seg_len_pow = ( int ) Math . pow ( 10, seg_len - 1 );
  int curr_val = res;
  for ( int i = 1;
  i <= ( s . length ( ) - seg_len );
  i ++ ) {
    curr_val = curr_val - ( s . charAt ( i - 1 ) - '0' ) * seg_len_pow;
    curr_val = curr_val * 10 + ( s . charAt ( i + seg_len - 1 ) - '0' );
    res = Math . max ( res, curr_val );
  }
  return res;
}

File: TransCoder_tokenized_test_set_functions/python/MAXIMUM_SEGMENT_VALUE_PUTTING_K_BREAKPOINTS_NUMBER.py
def findMaxSegment ( s, k ) :
    seg_len = len ( s ) - k
    res = 0
    for i in range ( seg_len ) :
        res = res * 10 + ( ord ( s [ i ] ) - ord ( '0' ) )
    seg_len_pow = pow ( 10, seg_len - 1 )
    curr_val = res
    for i in range ( 1, len ( s ) - seg_len ) :
        curr_val = curr_val - ( ord ( s [ i - 1 ] ) - ord ( '0' ) ) * seg_len_pow
        curr_val = ( curr_val * 10 + ( ord ( s [ i + seg_len - 1 ] ) - ord ( '0' ) ) )
        res = max ( res, curr_val )
    return res
